astropy_helpers/tests/test_git_helpers.py

```diff
@@ -37,6 +37,12 @@
 
 if not RELEASE:
     VERSION += get_git_devstr(False)
+    # This workaround is required to account for the fact that we test
+    # installation of this package in test_installed_git_version even with
+    # development versions. In these cases get_git_devstr does not work since
+    # it is called from outside of a git repository.
+    if VERSION.endswith('.dev'):
+        VERSION = VERSION.replace('.dev', '')
```

Comment:
If ``get_git_devstr`` doesn't work, shouldn't the version number retain at least the ``.dev`` suffix so it doesn't look like a stable version?

Reply:
I'm not sure it matters much since all of this is just within a test sandbox, but another possible solution would be to do something like this instead:
```python
if VERSION.endswith('.dev'):
    VERSION += '0'
```

Reply:
It does matter for developers. I have multiple environments with different dev versions. So, I would like that info retained as well.

Reply:
@pllim, agreed, but the only reason there is a problem here is because the test itself is running in a sandbox outside of a git tree, and this fix is solely intended to address that problem. This isn't an issue that a developer should ever experience directly, unless I'm missing something.

Reply:
I see it even when running from a `git` source checkout for some of my projects but I never bothered to investigate. Somehow the final `pytest` report still gave me the correct dev string. Anyway, if this version string is going into a pytest report, it is less confusing to have a "dev" still retained. Otherwise, people might think the test result is from a released version. Or am I misunderstanding something here?

Reply:
This could just as easily have been addressed in the call to `pytest.mark.parametrize` on line 190, by changing `'1.0.dev'` to `'1.0.dev0'` and getting rid of the call to `get_git_devstr` altogether. That might be cleaner, and maybe I should do it.

The problem with that approach is I guess we never actually test `get_git_devstr` in that case, but it's really not possible to test its proper behavior if the tests always run in a temp directory outside of the source tree.

Reply:
@drdavella - if someone downloads the dev version of astropy from git as a zip file they will end up (I think) with the full dev version but no ``.git`` directory.

Reply:
The dev string in this test will not be reported by pytest. This is part of a meta-test of installing a fake package and making sure astropy_helpers tags the installed version with the correct git hash.

When you and I run setup.py directly, it's always within a git tree. The setup.py under question here is generated as part of a fake package that is used for testing purposes. At one point in the test, a tarred up version of the package is extracted into a temporary directory. There's no longer a git tree here, which is why the 'dev' VERSION string doesn't get a revision number. I believe this was just an oversight from when the test was first designed, and my commit just tries to correct for it.

Reply:
Ohhh... Just the *test*... Sorry for the noise. I obviously need more coffee. Will promptly remedy that right now. ☕️

Reply:
@astrofrog, I guess that's a use case I hadn't considered. In that case, I'm not sure what the right solution should be. It's possible `get_git_devstr` could just return '0' in the absence of a git repository. That would avoid the warning from `setuptools`, but maybe in that case `get_git_devstr` should issue its own warning indicating that it's running outside of a git repo.

Reply:
> It's possible get_git_devstr could just return '0' in the absence of a git repository. That would avoid the warning from setuptools, but maybe in that case get_git_devstr should issue its own warning indicating that it's running outside of a git repo.

That sounds like the right approach